examples/todo-mvc/rust/src/ui/create_task_adapter.rs

```diff
@@ -1,89 +1,86 @@
 // Copyright © SixtyFPS GmbH <info@slint.dev>
 // SPDX-License-Identifier: MIT
 
+use std::rc::Rc;
+
 use slint::*;
 
 use crate::{
     mvc::{
-        {CreateTaskController, TaskListController}, {DateModel, TimeModel},
+        CreateTaskController, CreateTaskControllerCallbacks, DateModel, TaskListController,
+        TimeModel,
     },
     ui,
 };
 
-// a helper function to make adapter and controller connection a little bit easier
-fn connect_with_controller(
-    view_handle: &ui::MainWindow,
-    controller: &CreateTaskController,
-    connect_adapter_controller: impl FnOnce(ui::CreateTaskAdapter, CreateTaskController) + 'static,
-) {
-    connect_adapter_controller(view_handle.global::<ui::CreateTaskAdapter>(), controller.clone());
+pub fn create_controller_callbacks(view_handle: &ui::MainWindow) -> CreateTaskControllerCallbacks {
+    CreateTaskControllerCallbacks {
+        on_refresh: Box::new({
+            let view_handle = view_handle.as_weak();
+
+            move |create_task_model| {
+                ui::CreateTaskAdapter::get(&view_handle.unwrap())
+                    .set_title(create_task_model.title.into());
+                ui::CreateTaskAdapter::get(&view_handle.unwrap())
+                    .set_due_date(map_date_model_to_date(create_task_model.due_data));
+                ui::CreateTaskAdapter::get(&view_handle.unwrap())
+                    .set_due_time(map_time_model_to_time(create_task_model.due_time));
```

Comment:
```suggestion
                let Some(view) = view_handle.upgrade() else { return; };
                adapter = ui::CreateTaskAdapter::get(&view);
                adapter.set_title(create_task_model.title.into());
                adapter.set_due_date(map_date_model_to_date(create_task_model.due_data));
                adapter.set_due_time(map_time_model_to_time(create_task_model.due_time));
```

Same pattern in the other functions where `::get()` is called repeatedly.